functional/FunctionalTest.java

```diff
@@ -3815,9 +3847,9 @@ public static void main(String[] args) throws Exception {
 
     Process minioProcess = null;
 
-    String kmsKeyName = "my-minio-key";
+    String kmsKeyName = null;
     if (args.length != 4) {
-      endpoint = "http://localhost:9000";
+      endpoint = "https://localhost:9000";
```

Comment:
This tests should also run on non TLS setup to ensure the functionality.

Reply:
non-TLS is not necessary @balamurugana - production workloads are TLS based so just testing these should cover these cases. 

Since TLS based are specially SSE-C ones and SSE-KMS ones rest of them do not behave any differently either with TLS or without TLS. 

So I don't see we should have non-TLS based setup as well to be tested.

Reply:
NonTLS tests ensure code quality and correct functionality of `minio-java`. The code flow of TLS and NonTLS is different and we should not get any surprises in future. The server is in production or not is out of question for `minio-java`

Reply:
If you want to be very specific for certain features like sha256 calculation etc then we can enable it after this PR is merged. I will send a change @balamurugana

Reply:
Let's try to get this in atleast, AFAIK the functionality itself is complete in this PR and working fine with the latest MinIO releases.